processor/src/system/mod.rs

```diff
@@ -100,8 +101,8 @@ impl System {
 
     /// Returns free memory pointer at the specified clock cycle.
     #[inline(always)]
-    pub fn get_fmp_at(&self, clk: usize) -> Felt {
-        self.fmp_trace[clk]
+    pub fn get_fmp_at(&self, trace_len: usize) -> Felt {
+        self.fmp_trace[trace_len]
```

Comment:
I think we should keep the previous names here, since we're indexing by clock cycle and the clock cycle that's passed in may not be related to the trace length.